### ISSUE TITLE
Multi validator e2e's should use different patch release versions to detect app hash mismatch

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3663]
## Summary

Modify our e2e setup to use different patch releases to catch app hash mismatches

## Problem Definition

App hash mismatches are very subtle and hard to catch in reviews. Changing a store call from `Get` to `Has` may not seem state machine breaking, but it is since different amounts of gas are charged for each and `GasUsed` by a transaction is included in the block hash. Our e2e's only protect us against non-determinstic app hash bugs, but not deterministic bugs across patch versions

## Proposal

Modify the e2e setup to use a unique patch release. That is, specify a release line for the e2e test, such as `v4.4.x` and auto assign each validator a different patch version based on the available list. So if v4.4.x is used, at least one validator should have `v4.4.0` and another should have `v4.4.1`. This should allow us to catch app hash mismatches due to state machine breaking changes in newer patch releases

So form of this proposal is acceptable, I just gave one concrete solution as a starting point

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by chatton** [comment]
I think as a starting point, I think we could do something along the lines of

validator 0 is point release 0, validator 1 is point release 1 etc.

If there are more validators than point releases, we can just have the remaining validators use the last available release.

We can find tags [here](https://api.github.com/repos/cosmos/ibc-go/tags) from our E2Es. 

Determining what point release to use on PRs would be a very messy problem, I think we can start by just doing this only in compatibility tests.

**Comment by colin-axner** [comment]
Agreed! We should probably also add assertions that no validators have gone offline before closing out the tests (in cases enough validators don't crash to continue the network)

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
